react/OrderPlacedNotices.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { Fragment, FC } from 'react'
+import { FormattedMessage, FormattedDate } from 'react-intl'
+
+import ButtonLink from './components/ButtonLink'
+import FormattedPrice from './components/FormattedPrice'
+import { useOrderGroup } from './components/OrderGroupContext'
+import { parseBankInvoiceUrl, getPaymentInfoFromOrder } from './utils'
+import styles from './styles.css'
+
+const Notices: FC = () => {
+  const { orders, totalDeliveryParcels, totalPickUpParcels } = useOrderGroup()
+  const hasDelivery = totalDeliveryParcels.length > 0
+  const hasPickUp = totalPickUpParcels.length > 0
+
+  if (!hasDelivery && !hasPickUp) {
+    return null
+  }
+
+  const numOrders = orders.length
+  const isSplitOrder = numOrders > 1
+  const [bankInvoice] = orders
+    .map(getPaymentInfoFromOrder)
+    .filter(
+      paymentInfo =>
+        paymentInfo.paymentGroup === 'bankInvoice' && !!paymentInfo.url
+    )
+
+  const hasBankInvoice = bankInvoice != null
+  const listItems = [
```

Comment:
This `listItems` is so big, maybe writing a util function to do this logic would be better, wouldn't it?

Reply:
I agree that it's ugly and enormous right now, will try to improve it somehow. The previous implementation had all those logics and repeated enormous classnames in the markup.

Reply:
i tried....